Microsoft.HealthVault.Fhir/Transformers/ObservationToHealthVault.cs

```diff
@@ -32,39 +36,97 @@ public static class ObservationToHealthVault
         /// <returns>The HealthVault thing</returns>
         public static ThingBase ToHealthVault(this Observation observation)
         {
-            // ToDo: detect the type from the codeable values
-            return observation.ToWeight();
+            return observation.ToHealthVault(VocabToHealthVaultHelper.DetectHealthVaultTypeFromObservation(observation));
         }
 
-        private static Weight ToWeight(this Observation observation)
+        private static ThingBase ToHealthVault(this Observation observation, Type type)
```

Comment:
>private [](start = 8, length = 7)

nit: may be have public, internal and private methods .. so this function will go to bottom ..